src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -163,17 +176,14 @@ public void actionPerformed(ActionEvent e) {
         final String s = (String) JOptionPane.showInputDialog(
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
-        if (s != null && s.length() > 0) {
-          //if alphanumeric
-          if(s.matches("[a-zA-Z0-9]+")) {
-            clientContext.user.addUser(s);
-            UserPanel.this.getAllUsers(listModel);
-          } else {
-            JOptionPane.showMessageDialog(UserPanel.this,
-                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
-                "User Not Created",
-                JOptionPane.ERROR_MESSAGE);
-          }
+        if (validator.isValidUserName(s)) {
```

Comment:
can we just call `isVaidName()` in clientUser?  I am not sure which one will be more efficient. It seems to me that we compile that pattern once again by calling `TextValidator` constructor while in `ClientUser`, we have called it once.
we can do:
```
ClientUser usr = clientContext.user;
if( ! usr.isValidName(s) ){
    ....
}
else{
    usr.addUser(s);
}
```
this way, for sure we only trigger `TextValidator` once. What do you think?